pytuya/__init__.py

```diff
@@ -149,25 +149,58 @@ def __init__(self, dev_id, address, local_key=None, dev_type=None, connection_ti
 
         self.port = 6668  # default - do not expect caller to pass in
 
+        self.s = None # persistent socket
```

Comment:
Sorry, that part was not part of the pull request. You can see the state function of BulbDevice.
I am new to github and my mistake was to commit on the master branch instead of  a new branch. 

Concerning the persistent socket, in my opinion it is for testing purpose only to see if issue https://github.com/codetheweb/tuyapi/issues/84 can be fix and it seems so.
But this implementation cause some side effects that's why retry is needed.

I can do this change if you think it is good enough or we can close the request. I think a wrapper on top of the core library will be a better choice. I starting something close to that for domoticz: https://github.com/tixi/Domoticz-Tuya-SmartPlug-Plugin

Let me know. In any case, i will not be available the coming week

Reply:
I've done the same thing with commits myself, its easily done. I'd prefer we do these separately.

If you are feeling brave you could branch what you have (to save it) and then forcible remove the commits from history you don't want (I suspect but do not know if this will update the PR). A new PR is fine.

RE re-try, I've a preference for doing that outside too but I would be OK if it was added as an option.

I have my own version of the never-closed socket code too. I never had a need for it/benefit so I never ended up committing it. But it could be useful for diagnostics.